packages/flutter_tools/templates/app_shared/macos.tmpl/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme.tmpl

```diff
@@ -5,6 +5,26 @@
    <BuildAction
       parallelizeBuildables = "YES"
       buildImplicitDependencies = "YES">
+      {{#withSwiftPackageManager}}
```

Comment:
Maybe we should always add the pre-actions?  Does it hurt anything if it's not using SPM?  I don't think it would add much time (because the next time it runs in the build phases it will be almost a no-op based on the target inputs and outputs).  The less we can avoid fragmentation the better imo.

Reply:
So it's not exactly a no-op, the pre-action unfortunately does not pass the `EXPANDED_CODE_SIGN_IDENTITY` build setting so the input is considered changed because the codesigning changes. So it does add a little bit of time (~4 seconds)